test/helpers.js

```diff
@@ -0,0 +1,30 @@
+// Helper function to determine whether this browser is supported by 
+// impress.js or not. Copied from impress.js itself.
+var _impressSupported = function() {
+  var pfx = (function () {
+    var style = document.createElement('dummy').style,
+        prefixes = 'Webkit Moz O ms Khtml'.split(' '),
+        memory = {};
+    return function ( prop ) {
+      if ( typeof memory[ prop ] === "undefined" ) {
+        var ucProp  = prop.charAt(0).toUpperCase() + prop.substr(1),
+            props   = (prop + ' ' + prefixes.join(ucProp + ' ') + ucProp).split(' ');
+        memory[ prop ] = null;
+        for ( var i in props ) {
+            if ( style[ props[i] ] !== undefined ) {
+                memory[ prop ] = props[i];
+                break;
+            }
+        }
+      }
+      return memory[ prop ];
+    };
+  })();
+
+  var ua = navigator.userAgent.toLowerCase();
+  return   ( pfx("perspective") !== null ) &&
+           ( document.body.classList ) &&
+           ( document.body.dataset ) &&
+           ( ua.search(/(iphone)|(ipod)|(android)/) === -1 );
+}
```

Comment:
That's not really why this is here. This is related to testing that some CSS classes were set by impress.js code:

      if ( !_impressSupported() ) {
        assert.ok( notSupportedClass,
                   "body.impress-not-supported class still there." );
        assert.ok( !yesSupportedClass,
                   "body.impress-supported class was NOT added." );
      } else {
        assert.ok( !notSupportedClass,
                   "body.impress-not-supported class was removed." );
        assert.ok( yesSupportedClass,
                   "body.impress-supported class was added." );
                   
If we just hard-coded assume that the test is run in a browser where impressSupported is true, then it defeats the purpose of the whole test. If the test ever fails, you wouldn't know whether it is because the browser check is failing, or because the code is setting css class incorrectly.